checkpoint.go

```diff
@@ -141,12 +141,15 @@ func Checkpoint(w *wal.WAL, from, to int, keep func(id uint64) bool, mint int64)
 		}
 	}
 
-	cpdir := filepath.Join(w.Dir(), fmt.Sprintf("checkpoint.%06d", to))
-	cpdirtmp := cpdir + ".tmp"
+	cpdir := filepath.Join(w.Dir(), fmt.Sprintf(checkpointPrefix+"%06d", to))
+	cpdirtmp := cpdir + fileutil.TmpExt
 
 	if err := os.MkdirAll(cpdirtmp, 0777); err != nil {
 		return nil, errors.Wrap(err, "create checkpoint dir")
 	}
+	// Ensure that an early return doesn't leave any tmp folder.
+	defer os.RemoveAll(cpdirtmp)
```

Comment:
Pretty sure this will break in Windows (atleast for the test given below). When we open the `wal` below, the chances are there are open references to the files inside the directory that we are not closing. Don't have access to windows machine now to test this though.


Also don't have a good idea as to how to fix this.